### PR TITLE
feat: shortening prompt for expression classification.

### PR DESCRIPTION
**Pull request by racinmat**
The LLM option on character expressions is a great feature, but currently it sends the whole prompt into the LLM, which is usually quite large, and includes lots of information unnecessary for the classification itself.
This PR adds 2 new options:
- keeping only the conversation, skipping the world info, author's note, character description, scenario etc.
- using only last N messages from the conversation

I have performed experiments on Czech language using conversation with 12 responses from the character, and there using only the 3 last messages from the conversation, and skipping everything before conversation started almost did not degrade the classification quality and it shortened the number of tokens 20x times.
In short experiment with Seraphina, using only last 3 messages lowers the amount of tokens 4x, which is still great I think. 
In case of using paid LLM this makes the expression classification much more affordable.

### CODE REVIEW

**Comment by racinmat** [comment]
Notes from Cohee:
I can use interceptor, but it does not work for quiet prompts, that would be a breaking change:
https://github.com/SillyTavern/SillyTavern/blob/1.12.0/public/scripts/extensions/vectors/manifest.json#L8
https://github.com/SillyTavern/SillyTavern/blob/1.12.0/public/scripts/extensions/vectors/index.js#L568
https://github.com/SillyTavern/SillyTavern/blob/1.12.0/public/scripts/extensions/vectors/index.js#L478
This should be enough of references to get started
TL;DR:
- Interceptor added to manifest
- Interceptor registered in global scope (window)
- Interceptor implements a function that accepts a mutable chat array

text completion prompt can be modified using event handler:
Here you can modify a text completion prompt with an event handler (before combining):
https://github.com/SillyTavern/SillyTavern/blob/release/public/script.js#L3958

Chat Completions have a different event, but only after combining the prompt.

This feature surely requires a more careful planning. There's just too much to think about and things that may break in unexpected ways. But you know what, I think I have a solution for it.
Why not use a generateRaw and build the prompt yourself?
Instead of relying on ST's prompt builder and its quirks
You pull messages from a chat, add a system prompt, and here it is - your extended prompt.
Example from summary extension
https://github.com/SillyTavern/SillyTavern/blob/release/public/scripts/extensions/memory/index.js#L569
https://github.com/SillyTavern/SillyTavern/blob/release/public/scripts/extensions/memory/index.js#L598

**Comment by Cohee1207** [comment]
Closing because it requires rethinking with the use of generateRaw.
